scs2-simulation/src/main/java/us/ihmc/scs2/simulation/physicsEngine/contactPointBased/ContactPointBasedRobotPhysics.java

```diff
@@ -60,6 +77,14 @@ public ContactPointBasedRobotPhysics(RobotInterface owner)
       FrameShapePosePredictor frameShapePosePredictor = new FrameShapePosePredictor(forwardDynamicsCalculator);
       collidables.forEach(collidable -> collidable.setFrameShapePosePredictor(frameShapePosePredictor));
 
+      indexProvider = JointMatrixIndexProvider.toIndexProvider(owner.getAllJoints().toArray(new JointBasics[0]));
+      int nDoFs = indexProvider.getIndexedJointsInOrder().stream().map(joint -> joint.getDegreesOfFreedom()).reduce(0, Integer::sum);
+
+      String[] rowNames = getRowNames(owner, nDoFs);
+      jointsTauController = new YoMatrix("tau_control", nDoFs, 1, rowNames, null, owner.getRegistry());
+      jointsTauSimulationEffects = new YoMatrix("tau_sim", nDoFs, 1, rowNames, null, owner.getRegistry());
+      jointsTau = new YoMatrix("tau_total", nDoFs, 1, rowNames, null, owner.getRegistry());
```

Comment:
`owner.getRegistry()` means these go in the root `nadia` registry in an SCS2 session. We might want to avoid this clutter.

Reply:
There shouldn't be a need to create a whole new set of yoVariables for `jointTauController` as we expect the controller to eventually write its taus to the joints directly, so I'd either create a `YoMatrix` that points to the joint tau yovariables or not have that extra `YoMatrix`
For the registry, I think it's nice to have these extra yoVariables as it makes it explicit. We might wanna consider adding description for them. We can look into that when the PR is about done.

Reply:
Changed `jointTauController` to a `DMatrix`. Ok, I'm fine with having the YoMatrices `tau_llc` and `tau_total` in the `nadia` registry if you are. Not quite sure what you mean by adding description however.

Reply:
YoVariables can be constructed with a description. We rarely use that feature, but see for instance there's a construction in `YoDouble` that has a description: `YoDouble(String name, String description, YoRegistry registry)`. This description is used in the GUI to display a little tooltip when hovering the variable, can be useful to describe what the variable represents. I'm thinking we should add the option to specify a description in `YoMatrix` describe the tau_llc & tau_total, that'd probably be useful.